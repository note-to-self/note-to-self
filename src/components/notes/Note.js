import React from 'react';
import PropTypes from 'prop-types';
import styles from 'styled-components';

const NoteTitle = styles.h3`
@import url(‘https://fonts.googleapis.com/css?family=Muli:300,700’);
 font-family: ‘Muli’, sans-serif;
 color: #023B50;
`;

const NoteButton = styles.button`
 font-size: .8em;
`;

const NoteCard = styles.section`
 border: 1pt solid #8BCBC8;
 border-radius: 10px
 margin-bottom: 1em;
 padding: .3em;
 border: 1pt solid #8BCBC8;
 margin: 1em;

ul {
  list-style: none;
  padding: 0;
}
`;

export default function Note({ body, handleSubmit, schedule, buttonLabel, id }) {
  return (
    <NoteCard>
      <NoteTitle>{body}</NoteTitle>
      {schedule && <p>This is a {schedule} affirmation.</p>}
      {handleSubmit && <form onSubmit={handleSubmit.bind(null, id)}>
        <NoteButton>{buttonLabel} </NoteButton>
      </form>}
    </NoteCard>
  );
}

Note.propTypes = {
  body: PropTypes.string.isRequired,
  handleSubmit: PropTypes.func,
  schedule: PropTypes.string,
  buttonLabel: PropTypes.string,
  id: PropTypes.string.isRequired
};
