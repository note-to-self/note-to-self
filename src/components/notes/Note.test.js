
import React from 'react';
import { shallow } from 'enzyme';
import Note from './Note';

describe('Note snapshot', () => {
  it('matches a snapshot', () => {
    const handleSubmit = jest.fn();
    const handleCheckbox = jest.fn();
    const wrapper = shallow(
      <Note 
        body="You got this!"
        onSubmit={handleSubmit}
        schedule="daily"
        checkboxLabel="delete"
        checkboxValue={false}
        onChange={handleCheckbox}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
