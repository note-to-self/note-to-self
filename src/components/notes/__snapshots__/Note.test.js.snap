// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Note snapshot matches a snapshot 1`] = `
<styled.section>
  <styled.h3>
    You got this!
  </styled.h3>
  <p>
    This is a 
    daily
     affirmation.
  </p>
</styled.section>
`;
